### ISSUE TITLE
Current dev won't compile with the Debug build type

### ISSUE THREAD

**Issue by pordyna** [ComputationalRadiationPhysics/picongpu#4145]
Compiling the `Empty` example on the current dev branch with `pic-build -f -c "-DCMAKE_BUILD_TYPE=Debug"` results in:

```
[ 95%] Building CUDA object CMakeFiles/picongpu.dir/versionFormat.cpp.o
/home/ordyna35/src/picongpu-git/include/pmacc/../pmacc/mappings/kernel/StrideMapping.hpp: In member function ‘bool pmacc::StrideMapping<areaType, T_stride, baseClass<DIM, SuperCellSize_> >::next() [with unsigned int areaType = 3; unsigned int T_stride = 3; baseClass = pmacc::MappingDescription; unsigned int DIM = 3; SuperCellSize_ = pmacc::math::CT::Vector<mpl_::integral_c<int, 8>, mpl_::integral_c<int, 8>, mpl_::integral_c<int, 4> >]’:
/home/ordyna35/src/picongpu-git/include/pmacc/../pmacc/mappings/kernel/StrideMapping.hpp:114:1: error: inlining failed in call to ‘always_inline’ ‘bool pmacc::StrideMapping<areaType, T_stride, baseClass<DIM, SuperCellSize_> >::next() [with unsigned int areaType = 3; unsigned int T_stride = 3; baseClass = pmacc::MappingDescription; unsigned int DIM = 3; SuperCellSize_ = pmacc::math::CT::Vector<mpl_::integral_c<int, 8>, mpl_::integral_c<int, 8>, mpl_::integral_c<int, 4> >]’: recursive inlining
  114 |         HINLINE bool next()
      | ^   
/home/ordyna35/src/picongpu-git/include/pmacc/../pmacc/mappings/kernel/StrideMapping.hpp:126:12: note: called from here
  126 |                 return next();
      |            ^ 
make[2]: *** [CMakeFiles/picongpu.dir/main.cpp.o] Error 1
make[1]: *** [CMakeFiles/picongpu.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by sbastrakov** [comment]
Oh thanks for reporting. We always recommend against using debug build type, as everything takes forever (including compilation, even relatively to our usual times). But of course, it should still compile.

**Comment by sbastrakov** [comment]
Solved by #4147
